x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/YamlLifecyclePolicyConfig.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.template;
+
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.ilm.LifecyclePolicy;
+import org.elasticsearch.xpack.core.ilm.LifecyclePolicyUtils;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.elasticsearch.xpack.core.template.ResourceUtils.loadResource;
+
+public class YamlLifecyclePolicyConfig extends LifecyclePolicyConfig {
```

Comment:
Since we aren't defining this statically, it doesn't really look like this class is used at all, outside of the `load` method being called from `YamlTemplateRegistry.loadLifecyclePolicy` (which is only called from the constructor.

I think it doesn't make sense to have a generic type for it if it's all the function of returning a `LifecyclePolicy` object. I think we should collapse this class into the body of `YamlTemplateRegistry.loadLifecyclePolicy(String name)` and remove this class entirely